src/main/webapp/components/SearchBar.vue

```diff
@@ -27,36 +27,63 @@
 module.exports = {
   data: function() {
     return {
-      school: "",
-      course: "",
-      error: ""
+      school: this.$route.query.school,
+      course: this.$route.query.course,
+      error: null,
     }
   },
+  computed: {
+    formIsValid: function() {
+      return this.school || this.course;
+    },
+  },
   methods: {
     onSubmit: function(event) {
-      if(!this.school && !this.course) {
-        this.error = "Please enter at least one search criterion.";
-      }
-      else /* Form validated */ {
-        this.error = "";
-        this.queryDatabase(this.school, this.course);
+      if(this.formIsValid) {
+        this.onValidForm();
+      } else {
+        this.onInvalidForm();
       }
     },
-    queryDatabase: function(school, course) {
+    onValidForm: function () {
+      this.error = null;
+      this.$router.push({
+        path: 'search',
+        query: {
+          school: this.school,
+          course: this.course,
+        }
+      });
+    },
+    onInvalidForm: function () {
+      this.error = "Please enter at least one search criterion.";
+    },
+    fetchAndEmitResults: async function() {
       let url = new URL("/search", window.location.href);
-      url.searchParams.set('school', school);
-      url.searchParams.set('course', course);
+      this.school && url.searchParams.set('school', this.school);
```

Comment:
Perhaps I just don't know about this syntax, but why do you need to use this here when you're just calling a setter?

Reply:
I was having an issue where if the school was null, it would set the url search param for school=null, which the back end would read as school = "null" and then try to look for schools named null XD this makes sure null doesn't end up in the query string.

Reply:
oh interesting!